src/debuggability.js

```diff
@@ -178,6 +188,79 @@ Promise.prototype._propagateFrom = function (parent, flags) {
     USE(flags);
 };
 
+Promise.enableMonitoring = function () {
+    function registerPromise() {
+        if (Promise.monitor) {
+            Promise.monitor._promiseIdCounter++;
+            if (Promise.monitor._promiseIdCounter === Number.MAX_VALUE) {
+                Promise.monitor._promiseIdCounter = 0;
+            }
+            if (Promise.monitor._pendingPromises[
+                    Promise.monitor._promiseIdCounter]) {
+                // Use case when number of promises is higher than
+                // Number.MAX_VALUE and collision happens is not handled,
+                // disabling the monitoring feature.
+                // Probability of this case is very low
+                Promise.disableMonitoring();
+                throw new Error(
+                    "Promises ids collision happened, sorry." +
+                    " Monitoring feature will be disabled");
+            }
+            this._promiseId = Promise.monitor._promiseIdCounter;
+            Promise.monitor._pendingPromises[Promise.monitor._promiseIdCounter]
+                = this;
+        }
+    }
+
+    function unregisterPromise() {
+        if (Promise.monitor)
+            delete Promise.monitor._pendingPromises[this._promiseId];
+    }
+
+    if (!Promise.monitor) {
+        // Property that holds monitoring related info,
+        // existence of it means that monitoring feature is currently enabled
+        Promise.monitor = {};
+        Promise.monitor._pendingPromises = {};
+        Promise.monitor._promiseIdCounter = 0;
+
+        Promise.disableMonitoring = function () {
+            // No reason to clean up the id's from pending promises
+            util.unhookFrom(Promise.prototype, "_promiseCreated",
+                registerPromise);
+            util.unhookFrom(Promise.prototype,
+                "_promiseSettled", unregisterPromise);
+            delete Promise.monitor;
```

Comment:
Is `delete` still performance-sensitive? Usually overwriting by `null` is enough.